chain/network/src/network_protocol.rs

```diff
@@ -424,3 +424,27 @@ impl PeerMessage {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    macro_rules! assert_size {
+        ($type:ident, $allowed_size: expr) => {
+            let struct_size = std::mem::size_of::<$type>();
+            assert!(
+                struct_size <= $allowed_size,
+                "The size of {} is {}",
+                stringify!($type),
+                struct_size
+            )
+        };
+    }
+
+    #[test]
+    fn test_assert_size() {
+        assert_size!(HandshakeFailureReason, 64);
+        assert_size!(PeerMessage, 1144);
+        #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+        assert_size!(RoutingSyncV2, 80);
+    }
+}
```

Comment:
It might make sense to use `static_assert` crate here.

Reply:
@matklad The best way would be to use `1.57.0` features 
```const _: () = assert!(std::mem::size_of::<u64>() == 8);```

Unfortunately, `const_format` is not fully stable yet. https://docs.rs/const_format/latest/const_format/
I can write a static assert like this:
```
assert!(std::mem::size_of::<u64>() == 8, "u64");
```
However, `assert!` doesn't support formatting in `static` case on rust stable.
I think it's best to add `TODO`, once that feature becomes stable.

Reply:
Yeah, I mean that we can use this crate https://crates.io/crates/static_assertions which we already use, and which contains all the necessary polyfils. 

It was possible to "hand-write" this even before 1.57 (`[()][!cond]`), and I generally do perfer hand-writing such simple utilities, but, as we *already* depend on the crate, we might as use it. Though, no strong opinion here, the current approach is also fine!

Reply:
@matklad I looked into https://docs.rs/static_assertions/latest/static_assertions/
However, it's functionally the same as `1.57.0` static asserts. Neither of those examples contains proper prints on assert with format.

Reply:
Ok, I changed the code to use static asserts from `1.57.0`, they look good.

Reply:
https://doc.rust-lang.org/nightly/std/macro.const_format_args.html#